pymemcache/client/base.py

```diff
@@ -307,6 +307,7 @@ def set(self, key, value, expire=0, noreply=None):
                   from the cache, or zero for no expiry (the default).
           noreply: optional bool, True to not wait for the reply (defaults to
                    self.default_noreply).
+          flags: int, arbitrary bit field used for server-specific flags
```

Comment:
the original logic is, if a serializer is specified when constructing the client, it will generate the flags using the serializer, otherwise flags will be set to 0 as default. 
Current logic won't change that part of logic, it will still generate flags if serializer is specified while constructing the client, or default to 0 if not specified. 
For the extension logic, we do not specify serializer, so it would not use that piece of logic. We will pass flags for the set function. 
This will not affect the current usage of set function, the code worked before could still work with the new logic.

Reply:
(I think this was a reply to the latter comment, so I'll move my reply there.)